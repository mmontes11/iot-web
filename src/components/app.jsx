import React, { Fragment } from "react";
import { Provider } from "react-redux";
import PropTypes from "prop-types";
import "styles/index.scss";

const App = ({ store }) => (
  <Provider store={store}>
    <Fragment>{null}</Fragment>
  </Provider>
);

App.propTypes = {
  store: PropTypes.shape({}).isRequired
};

export default App;
