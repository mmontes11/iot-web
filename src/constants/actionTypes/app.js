export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
