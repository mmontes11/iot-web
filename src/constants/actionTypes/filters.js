export const FILTER_TYPE_SELECT = "FILTER_TYPE_SELECT";
export const ADD_FILTER_TYPE = "ADD_FILTER_TYPE";
