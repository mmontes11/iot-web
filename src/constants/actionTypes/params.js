export const TYPE_SELECT = "TYPE_SELECT";
export const TYPE_UPDATED = "TYPE_UPDATED";
export const OBSERVATION_SELECT = "OBSERVATION_SELECT";
export const OBSERVATION_UPDATED = "OBSERVATION_UPDATED";
export const OBSERVATIONS_REQUEST = "OBSERVATIONS_REQUEST";
export const OBSERVATIONS_REQUEST_SUCCESS = "OBSERVATIONS_REQUEST_SUCCESS";
export const OBSERVATIONS_REQUEST_ERROR = "OBSERVATIONS_REQUEST_ERROR";
export const OBSERVATIONS_UPDATED = "OBSERVATIONS_UPDATED";
