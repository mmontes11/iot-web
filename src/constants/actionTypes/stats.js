export const TYPE_SELECT = "TYPE_SELECT";
export const TYPE_UPDATED = "TYPE_UPDATED";