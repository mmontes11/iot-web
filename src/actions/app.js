import { TOGGLE_HAMBURGER_MENU, TOGGLE_MAP_DIALOG, SET_SHOW_ERROR } from "constants/actionTypes/app";

export const toggleHamburgerMenu = () => dispatch => {
  dispatch({ type: TOGGLE_HAMBURGER_MENU });
};

export const toggleMapDialog = () => dispatch => {
  dispatch({ type: TOGGLE_MAP_DIALOG });
};

export const setShowError = showError => dispatch => {
  dispatch({ type: SET_SHOW_ERROR, showError });
};
