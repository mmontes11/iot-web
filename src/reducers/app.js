import { TOGGLE_HAMBURGER_MENU, TOGGLE_MAP_DIALOG } from "constants/actionTypes/app";

const initialState = {
  isHamburgerMenuExpanded: false,
  isMapDialogOpened: false,
};

export default (state = initialState, { type }) => {
  switch (type) {
    case TOGGLE_HAMBURGER_MENU:
      return { ...state, isHamburgerMenuExpanded: !state.isHamburgerMenuExpanded };
    case TOGGLE_MAP_DIALOG:
      return { ...state, isMapDialogOpened: !state.isMapDialogOpened };
    default:
      return state;
  }
};
