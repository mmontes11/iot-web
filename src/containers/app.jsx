import React, { Fragment } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { compose } from "recompose";
import PropTypes from "prop-types";
import "styles/index.scss";
import * as fromState from "reducers";
import * as appActions from "actions/app";
import Login from "containers/login";
import Main from "components/main";
import Modal from "components/modal";

const App = ({ isAuth, shouldShowError, setShowError }) => (
  <Fragment>
    {isAuth ? <Main /> : <Login />}
    <Modal
      isActive={shouldShowError}
      onCloseClick={() => setShowError(false)}
      messageStyle="is-danger"
      title="Error"
      subTitle="Request failed"
    />
  </Fragment>
);

App.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  shouldShowError: PropTypes.bool.isRequired,
  setShowError: PropTypes.func.isRequired,
};

const withConnect = connect(
  state => ({
    isAuth: state.auth.isAuth,
    shouldShowError: fromState.shouldShowError(state),
  }),
  { setShowError: appActions.setShowError },
);

export default compose(
  withRouter,
  withConnect,
)(App);
