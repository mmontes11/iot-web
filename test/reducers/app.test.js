import appReducer from "reducers/app";
import { TOGGLE_HAMBURGER_MENU } from "constants/actionTypes/app";

describe("reducers/app", () => {
  const initialState = {
    isHamburgerMenuExpanded: false,
  };
  it("has a default state", () => {
    expect(appReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(appReducer(initialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces TOGGLE_HAMBURGUER_MENU", () => {
    expect(appReducer(initialState, { type: TOGGLE_HAMBURGER_MENU })).toMatchSnapshot();
  });
});
