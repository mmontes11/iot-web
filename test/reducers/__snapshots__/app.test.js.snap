// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/app has a default state 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
}
`;

exports[`reducers/app reduces TOGGLE_HAMBURGUER_MENU 1`] = `
Object {
  "isHamburgerMenuExpanded": true,
}
`;

exports[`reducers/app reduces nothing 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
}
`;
